core/src/main/scala/sttp/tapir/server/interpreter/ContentNegotiation.scala

```diff
@@ -0,0 +1,93 @@
+package sttp.tapir.server.interpreter
+
+import sttp.model.{Header, HeaderNames, MediaType}
+import sttp.tapir.EndpointOutput.StatusMapping
+
+private[interpreter] class MediaTypeNegotiator(headers: Seq[Header]) {
+
+  private val acceptedMediaTypes: Seq[(MediaType, Float)] =
+    ContentNegotiation.extract(HeaderNames.Accept, headers)(MediaType.unsafeParse).sortBy {
```

Comment:
This is a good start, but I think the more natural place for this would be in sttp-model. We already have there a couple of classes which parse specific header values (cache directives, etags). Similarly, we could have a parser for the `Accept` header value, which would include the q values & the sorting logic.

We could also add some methods to `MediaType` to check if the main/sub types is a wildcard `*` to avoid these constants here

Reply:
I opened a PR in sttp-model https://github.com/softwaremill/sttp-model/pull/76